src/posix/platform/netif.cpp

```diff
@@ -1563,6 +1564,79 @@ static void processNetifInfoEvent(otInstance *aInstance, struct rt_msghdr *rtm)
 
 #endif
 
+#if defined(__linux__)
+
+#define ERR_RTA(errmsg, requestPayloadLength) \
+    ((struct rtattr *)((char *)(errmsg)) + NLMSG_ALIGN(sizeof(struct nlmsgerr)) + NLMSG_ALIGN(requestPayloadLength))
+
+// The format of NLMSG_ERROR is described below:
+//
+// ----------------------------------------------
+// | struct nlmsghdr - response header          |
+// ----------------------------------------------------------------
+// |    int error                               |                 |
+// ---------------------------------------------| struct nlmsgerr |
+// | struct nlmsghdr - original request header  |                 |
+// ----------------------------------------------------------------
+// | ** optionally (1) payload of the request   |
+// ----------------------------------------------
+// | ** optionally (2) extended ACK attrs       |
+// ----------------------------------------------
+//
+static void HandleNetlinkResponse(struct nlmsghdr *msg)
+{
+    const struct nlmsgerr *err = reinterpret_cast<const nlmsgerr *>(NLMSG_DATA(msg));
+    size_t                 rtaLength;
+    size_t                 requestPayloadLength = 0;
+    uint32_t               requestSeq           = err->msg.nlmsg_seq;
+
+    if (msg->nlmsg_len < NLMSG_LENGTH(sizeof(struct nlmsgerr)))
+    {
+        otLogWarnPlat("[netif] Truncated netlink reply of request#%u", requestSeq);
+        ExitNow();
+    }
+
+    // The payload of the request is omitted if NLM_F_CAPPED is set
+    if (!(msg->nlmsg_flags & NLM_F_CAPPED))
+    {
+        requestPayloadLength = NLMSG_PAYLOAD(&err->msg, 0);
+    }
+
+    rtaLength = NLMSG_PAYLOAD(msg, sizeof(struct nlmsgerr)) - requestPayloadLength;
+
+    if (err->error == 0)
```

Comment:
Is it better to do this check after we verified the message is not truncated?

Reply:
I think it is already the case. We will `ExitNow()` if we verified that this message is truncated before this line.

Reply:
I meant we can verify and exit at line 1600, no need to check NLM_F_CAPPED when `err->error` is `0`.

Reply:
Got you. Updated in the new push